plugin/src/index.ts

```diff
@@ -69,9 +69,15 @@ The plugin no longer uses this and it should be deleted to avoid conflicts.\n`)
     console.log('Creating site data metadata file')
     await createMetadataFileAndCopyDatastore(PUBLISH_DIR, cacheDir)
   }
-
-  await writeFunctions({ constants, netlifyConfig, neededFunctions })
-
+  // Gatsby V2 does not have functions so we can skip this step
+  if (neededFunctions.length !== 0) {
+    const helperFunctions = await import('./helpers/functions')
+    await helperFunctions.writeFunctions({
+      constants,
+      netlifyConfig,
+      neededFunctions,
+    })
+  }
```

Comment:
I wonder whether we can avoid importing `deleteFunctions` at the top of the file and dynamically import it here too. I'd assume that we won't need to delete functions if we also never need to create them.

Alternatively, we could look at where `gastbyFunction` is imported and attempt to fix it there (at the point where the problem exists).

We could avoid importing the `gatsbyFunction` module here:

https://github.com/netlify/netlify-plugin-gatsby/blob/main/plugin/src/templates/handlers.ts#L22

We'd then need to mock out the call to `gatsbyFunction` on this line:

https://github.com/netlify/netlify-plugin-gatsby/blob/main/plugin/src/templates/handlers.ts#L159

It would have to be a function which matches the signature of the function to keep the build process happy but it's only ever called by the serverless function.

At runtime it would still work as expected as the serverless function imports it here:

https://github.com/netlify/netlify-plugin-gatsby/blob/main/plugin/src/templates/handlers.ts#L186

@nickytonline @taty2010 - This might be overly complex but it potentially fixes it at the point where the function is needed. Thoughts?

Reply:
@ascorbic - Tagging you for input too. Feels like a potentially complex solution but could work?

Reply:
I agree with @pieh and wonder whether we've changed the functionality here. I'm not sure how we'd solve the dynamic import issue here if we don't do it like this. If we always need to call `deleteFunctions` then we'll always hit the dependency issue without a refactor of the way we generate the runtime function code as part of the build. 🤔